api/src/auth/api_jwt_auth.py

```diff
@@ -205,3 +206,12 @@ def example_method(db_session: db.Session) -> response.ApiResponse:
         # The message is just the value we set when constructing the JwtValidationError
         logger.info("JWT Authentication Failed for provided token", extra={"auth.issue": e.message})
         raise_flask_error(401, e.message)
+
+
+def set_token_expiration_time(config: ApiJwtConfig | None = None) -> datetime:
```

Comment:
I'd just make this method take in the user token session and set it in here.

I might also suggest naming it something like `refresh_token_expiration`.